core/src/mender-artifact.c

```diff
@@ -1152,6 +1152,20 @@ artifact_read_data_prepare(mender_artifact_ctx_t *ctx, mender_artifact_download_
         return MENDER_FAIL;
     }
 
+    size_t file_name_length = strlen(ctx->file.name);
+    if ((strlen("data/xxxx.tar.xx") == file_name_length) || (strlen("data/xxxx.tar.xxx") == file_name_length)) {
```

Comment:
What about things like `.tgz` and similar?

Reply:
The artifact [documentation](https://github.com/mendersoftware/mender-artifact/blob/master/Documentation/artifact-format-v3.md#compression) only state that `xz`, `zt`, and `zst` are supported.

Reply:
Worth adding a comment then.